subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -231,6 +237,18 @@ inline fun <T> withRootDisposable(action: Disposable.() -> T): T {
 }
 
 
+private
+inline fun withSystemProperty(propertyName: String, propertyValue: String, action: () -> Unit) {
```

Comment:
:x: should reuse `SystemProperties.withSystemProperty(name, value, block)` instead

Reply:
I considered that, but it's non inlineable.

Reply:
Done. I see the benefit now.